### PR TITLE
Crypto functions placed on top of the module and work with ursa signer encapsulated into Signature.

### PR DESCRIPTION
**Pull request by humb1t**
Signed-off-by: Nikita Puzankov <humb1t@yandex.ru>

### CODE REVIEW

File: iroha/src/crypto.rs [hunk 1]
```diff
@@ -5,26 +5,22 @@ use ursa::{
         digest::{Input, VariableOutput},
         VarBlake2b,
     },
-    keys::PublicKey as UrsaPublicKey,
-    signatures::{ed25519::Ed25519Sha512, SignatureScheme},
+    keys::{PrivateKey, PublicKey as UrsaPublicKey},
```
Comment (eadventurous) [inline iroha/src/crypto.rs:8]: Maybe rename `PrivateKey` to `UrsaPrivateKey` to be consistent with `UrsaPublicKey` import

Comment (humb1t) [reply iroha/src/crypto.rs:8]: I think we need to add our PrivateKey first. Let's discuss it in RFC
https://wiki.hyperledger.org/display/iroha/Encapsulation+of+crypto+related+functionality

File: iroha/src/crypto.rs [hunk 2]
```diff
@@ -37,11 +33,20 @@ pub struct Signature {
...
 
 impl Signature {
-    pub fn new(public_key: PublicKey, signature: Ed25519Signature) -> Signature {
-        Signature {
+    pub fn new(
+        public_key: PublicKey,
+        payload: &[u8],
+        private_key: &PrivateKey,
+    ) -> Result<Signature, String> {
+        let transaction_signature = Signer::new(&Ed25519Sha512, &private_key)
```
Comment (takemiyamakoto) [inline iroha/src/crypto.rs:41]: Ed25519Sha512 is fine for our MVP, but long term, we should allow people to set any crypto supported by URSA

Comment (humb1t) [reply iroha/src/crypto.rs:41]: https://soramitsu.atlassian.net/browse/RH2-71

**Pull request merged**